---
title: "Milestone"
author: "Wenyi Shao"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Background
I picked these two papers because they both use human breast cancer cell line MCF10-A. The goal is to compare whether the knockdown method for METTL3 will affect gene expression.


|   |   |
|---|---|
| Paper 1  | Paper2  |
|  CRISPR |  siRNA |
| Ribodepletion  |  poly-A |
| Read length 101  | Read length 150  |
|  Paired end | Paired end  |
| Stranded  |  Unstranded |
| 3 controls + 3 KD  |  2 controls + 4 KD (2 different siRNA) |

## Data Download & Preprocessing

### Reference genome and data download:
```{bash download, eval=FALSE}
# reference genome
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/genes/hg38.ncbiRefSeq.gtf.gz
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/hg38.fa.gz
# data for paper1 (CRISPR METTL3 knockdown)
accession_numbers=(SRR25236333 SRR25236334 SRR25236336 SRR25236337 SRR25236338 SRR25236339)

for accession_number in "${accession_numbers[@]}"; do
    fastq-dump --split-files --gzip "$accession_number"
done
# data for paper2 (siRNA METTL3 knockdown)
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/000/ERR9627500/ERR9627500_1.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/000/ERR9627500/ERR9627500_2.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/001/ERR9627501/ERR9627501_1.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/001/ERR9627501/ERR9627501_2.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/008/ERR9627488/ERR9627488_2.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/008/ERR9627488/ERR9627488_1.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/009/ERR9627489/ERR9627489_2.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/009/ERR9627489/ERR9627489_1.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/004/ERR9627494/ERR9627494_1.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/004/ERR9627494/ERR9627494_2.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/005/ERR9627495/ERR9627495_1.fastq.gz
wget -nc ftp://ftp.sra.ebi.ac.uk/vol1/fastq/ERR962/005/ERR9627495/ERR9627495_2.fastq.gz
```
**The names of the files are then manually changed. Steps of making new directories and re-directing to these directories are omitted.**

### Run fastQC on raw reads.
```{bash preQC, eval=FALSE}
for file in *.fastq.gz; do
        fastqc "$file" --extract -o fastqcResults/
done
```

### Generate STAR index for both papers
```{bash gen_index, eval=FALSE}
gunzip hg38.fa.gz
gunzip hg38.knownGene.gtf.gz
# paper1 has read length of 101
STAR --runMode genomeGenerate --runThreadN 1 --genomeDir hg38_STARindex/ --genomeFastaFiles hg38.fa --sjdbGTFfile hg38.knownGene.gtf --sjdbOverhang 99
# paper2 has read length of 150
STAR --runMode genomeGenerate --runThreadN 4 --genomeDir hg38_150/ --genomeFastaFiles hg38.fa --sjdbGTFfile hg38.ncbiRefSeq.gtf --sjdbOverhang 149 
```

### STAR alignment
To find min and max intron size:
```{bash intron, eval = FALSE}
wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/bigZips/genes/hg38.ncbiRefSeq.gtf.gz
# then created bed file and upload from local
scp /root/angsd/finalProject/hg38.bed cayuga:/athena/angsd/scratch/wes4004/finalProj
hg38.bed
awk '{print $3 - $2} ' hg38.bed | sort -n | uniq | head -n 3
awk '{print $3 - $2} ' hg38.bed | sort -rn | uniq | head -n 3
```
output for the smallest intron:
```
1
2
10
```
output for largest intron:
```
1160411
1097903
1068357
```

### Run STAR
```{bash runSTAR, eval=FALSE}
# Paper 1
for ((i=1; i<=3; i++)); do
        STAR --runMode alignReads \
        --runThreadN 4 \
        --genomeDir /athena/angsd/scratch/wes4004/finalProj/hg38_100 \
        --readFilesIn /athena/angsd/scratch/wes4004/finalProj/paper1/trimResults/"CRISPR_CTRL_${i}_1_val_1.fq.gz" "CRISPR_CTRL_${i}_2_val_2.fq.gz" \
        --readFilesCommand zcat \
        --outFileNamePrefix /athena/angsd/scratch/wes4004/finalProj/paper1/trimResults/STARalign/"CRISPR_CTRL_${i}." \
        --outSAMtype BAM SortedByCoordinate \
        --outFilterMultimapNmax 1 \
        --alignIntronMin 1 \
        --alignIntronMax 1200000 \
        --outSAMattributes NH HI AS nM MD
done

for ((i=1; i<=3; i++)); do
        STAR --runMode alignReads \
        --runThreadN 4 \
        --genomeDir /athena/angsd/scratch/wes4004/finalProj/hg38_100 \
        --readFilesIn /athena/angsd/scratch/wes4004/finalProj/paper1/trimResults/"CRISPR_KD_${i}_1_val_1.fq.gz" "CRISPR_KD_${i}_2_val_2.fq.gz" \
        --readFilesCommand zcat \
        --outFileNamePrefix /athena/angsd/scratch/wes4004/finalProj/paper1/trimResults/STARalign/"CRISPR_KD_${i}." \
        --outSAMtype BAM SortedByCoordinate \
        --outFilterMultimapNmax 1 \
        --alignIntronMin 1 \
        --alignIntronMax 1200000 \
        --outSAMattributes NH HI AS nM MD
done

# Paper 2
names=("siRNA_CTRL_1"
        "siRNA_CTRL_2"
        "siRNA_KD1_1"
        "siRNA_KD1_2"
        "siRNA_KD2_1"
        "siRNA_KD2_2")

for ((i = 0; i < ${#names[@]}; i++)); do
        file1="${names[i]}_1_val_1.fq.gz"
        file2="${names[i]}_2_val_2.fq.gz"

        STAR --runMode alignReads \
        --runThreadN 4 \
        --genomeDir /athena/angsd/scratch/wes4004/finalProj/hg38_150 \
        --readFilesIn /athena/angsd/scratch/wes4004/finalProj/paper2/trimResults/"$file1" "$file2" \
        --readFilesCommand zcat \
        --outFileNamePrefix /athena/angsd/scratch/wes4004/finalProj/paper2/trimResults/STARalign/"${names[i]}." \
        --outSAMtype BAM SortedByCoordinate \
        --outFilterMultimapNmax 1 \
        --alignIntronMin 1 \
        --alignIntronMax 1200000 \
        --outSAMattributes NH HI AS nM MD

done
```

### Index bam files
```{bash index, eval=FALSE}
for file in *.bam; do
        samtools index "$file"
done
```

## Alignment QC 

### Running featureCounts
```{bash featureCounts, eval=FALSE}
# paper 1 genes
featureCounts -a /athena/angsd/scratch/wes4004/finalProj/hg38.ncbiRefSeq.gtf -g gene_id -o paper1_counts.txt -p *.bam
# paper 1 exons 
featureCounts -a /athena/angsd/scratch/wes4004/finalProj/hg38.ncbiRefSeq.gtf -g gene_id -o paper1_counts.txt -p -f -O *.bam

# paper 2 genes
featureCounts -a /athena/angsd/scratch/wes4004/finalProj/hg38.ncbiRefSeq.gtf -g gene_id -o paper2_counts.txt -p *.bam
# paper2 exons
featureCounts -a /athena/angsd/scratch/wes4004/finalProj/hg38.ncbiRefSeq.gtf -g gene_id -o paper2_counts.txt -p -f -O *.bam
```

### Analyzing featureCounts results in R
```{r library, message=FALSE, warning=FALSE}
library(ggplot2)
library(dplyr)
library(tidyr)
```

```{r files}
paper1_exons <- read.table("C:/Users/hwenn/OneDrive/Desktop/邵文奕/学校/angsd/final project/paper1_exons.txt.summary", header = TRUE)
paper1_genes <- read.table("C:/Users/hwenn/OneDrive/Desktop/邵文奕/学校/angsd/final project/paper1_genes.txt.summary", header = TRUE)
paper2_exons <- read.table("C:/Users/hwenn/OneDrive/Desktop/邵文奕/学校/angsd/final project/paper2_exons.txt.summary", header = TRUE)
paper2_genes <- read.table("C:/Users/hwenn/OneDrive/Desktop/邵文奕/学校/angsd/final project/paper2_genes.txt.summary", header = TRUE)
```

```{r paper1}
orig_names_1 <- names(paper1_exons)

names(paper1_exons) <- c(names(paper1_exons)[1], "CRISPR_CTRL_1", "CRISPR_CTRL_2", "CRISPR_CTRL_3", "CRISPR_KD_1", "CRISPR_KD_2", "CRISPR_KD_3")
df_long_1 <- gather(paper1_exons, key = "Sample", value = "Value", -Status)
df_long_mod_1 <- df_long_1 %>% filter(Status %in% c("Assigned", "Unassigned_NoFeatures", "Unassigned_Ambiguity"))
df_long_mod_1$filter <- "exons"

names(paper1_genes) <- c(names(paper1_genes)[1], "CRISPR_CTRL_1", "CRISPR_CTRL_2", "CRISPR_CTRL_3", "CRISPR_KD_1", "CRISPR_KD_2", "CRISPR_KD_3")
df_long_gene_1 <- gather(paper1_genes, key = "Sample", value = "Value", -Status)
df_long_mod_gene_1 <- df_long_gene_1 %>% filter(Status %in% c("Assigned", "Unassigned_NoFeatures", "Unassigned_Ambiguity"))
df_long_mod_gene_1$filter <- "genes"

paper1 <- rbind(df_long_mod_1, df_long_mod_gene_1)

plot1 <- ggplot(paper1, aes(x = Sample, y = Value, fill = Status)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_y_continuous(labels = scales::scientific) +
  coord_flip() +
  labs(x = "Samples", y = "# Reads", title = "Exon Counts from FeatureCounts") +
   facet_grid(filter ~ .)
plot1

```


```{r paper2}
orig_names_2 <- names(paper2_exons)

names(paper2_exons) <- c(names(paper2_exons)[1], "siRNA_CTRL_1", "siRNA_CTRL_2", "siRNA_KD1_1", "siRNA_KD1_2", "siRNA_KD2_1", "siRNA_KD2_2")
df_long_2 <- gather(paper2_exons, key = "Sample", value = "Value", -Status)
df_long_mod_2 <- df_long_2 %>% filter(Status %in% c("Assigned", "Unassigned_NoFeatures", "Unassigned_Ambiguity"))
df_long_mod_2$filter <- "exons"

names(paper2_genes) <- c(names(paper2_genes)[1], "siRNA_CTRL_1", "siRNA_CTRL_2", "siRNA_KD1_1", "siRNA_KD1_2", "siRNA_KD2_1", "siRNA_KD2_2")
df_long_gene_2 <- gather(paper2_genes, key = "Sample", value = "Value", -Status)
df_long_mod_gene_2 <- df_long_gene_2 %>% filter(Status %in% c("Assigned", "Unassigned_NoFeatures", "Unassigned_Ambiguity"))
df_long_mod_gene_2$filter <- "genes"

paper2 <- rbind(df_long_mod_2, df_long_mod_gene_2)

plot2 <- ggplot(paper2, aes(x = Sample, y = Value, fill = Status)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_y_continuous(labels = scales::scientific) +
  coord_flip() +
  labs(x = "Samples", y = "# Reads", title = "Exon Counts from FeatureCounts") +
   facet_grid(filter ~ .)
plot2
```

### Run QoRTs
```{bash qorts, eval=FALSE}
# paper 1
for file in *.bam; do

        filename=$(basename "$file")
        filename_no_ext="${filename%%.*}"

        qorts -Xmx40G QC \
        --generatePlots \
        --maxReadLength 101 \
        --outfilePrefix "$filename_no_ext"_ \
        $file /athena/angsd/scratch/wes4004/finalProj/hg38.ncbiRefSeq.gtf ./qortsResults
done

# paper 2
for file in *.bam; do

        filename=$(basename "$file")
        filename_no_ext="${filename%%.*}"

        qorts -Xmx40G QC \
        --generatePlots \
        --maxReadLength 150 \
        --outfilePrefix "$filename_no_ext"_ \
        $file /athena/angsd/scratch/wes4004/finalProj/hg38.ncbiRefSeq.gtf ./qortsResults
done
```

### Running multiQC
The resulting .html file is in the repository.
```{bash multiqc, eval=FALSE}
multiqc finalProj/
```
