---
title: "Milestone 3"
author: "Wenyi Shao"
date: "`r Sys.Date()`"
output: 
  html_document: 
    toc: true
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Load libraries and files

### Load libraries
```{r library, message=FALSE, warning=FALSE}
library(tidyverse)
library(ggplot2); theme_set(theme_bw(base_size = 16))
library(magrittr)
library(DESeq2)
library(tidySummarizedExperiment)
options("restore_SummarizedExperiment_show" = TRUE)
library(pheatmap)
```

### Load files
```{r files}
paper1_genes <- read.table("C:/Users/hwenn/OneDrive/Desktop/邵文奕/学校/angsd/final project/paper1_counts.txt", header = TRUE)
orig_names_1_genes <- names(paper1_genes)
names(paper1_genes) <- c(names(paper1_genes)[1:6], "CRISPR_CTRL_1", "CRISPR_CTRL_2", "CRISPR_CTRL_3", "CRISPR_KD_1", "CRISPR_KD_2", "CRISPR_KD_3")

paper2_genes <- read.table("C:/Users/hwenn/OneDrive/Desktop/邵文奕/学校/angsd/final project/paper2_counts.txt", header = TRUE)

orig_names_2_genes <- names(paper2_genes)
names(paper2_genes) <- c(names(paper2_genes)[1:6], "siRNA_CTRL_1", "siRNA_CTRL_2", "siRNA_KD_1", "siRNA_KD_2", "siRNA_KD_3", "siRNA_KD_4")

genes <- cbind(paper1_genes, paper2_genes[, 7:12])
rm(paper1_genes, paper2_genes)
```

## DESeq object

### Create DESeq object
```{r deseqObj}
row.names(genes) <- make.names(genes$Geneid)
cts_genes <- as.matrix(genes[, -c(1:6)])
genes_coldata <- data.frame(condition = gsub("_[0-9]+", "", colnames(cts_genes)), row.names = colnames(cts_genes))

dds_genes <- DESeqDataSetFromMatrix(countData = cts_genes, colData = genes_coldata, design = ~ condition)

genes_rowdata <- genes[, 1:6]
rowData(dds_genes) <- genes_rowdata
```

### View gene counts of raw data
```{r counts}
head(assay(dds_genes, "counts"))

gene_sums <- colSums(counts(dds_genes))
gene_sums_df <- data.frame(Category = names(gene_sums), Count = gene_sums)
ggplot(gene_sums_df, aes(x = Category, y = Count)) +
  geom_bar(stat = "identity", alpha = 0.7) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = "Treatment", y = "Gene Count") +
  theme(plot.margin = margin(t = 20, r = 10, b = 10, l = 10, unit = "pt"))
```

### View size factor
```{r sizeFactor}
keep_genes <- rowSums(counts(dds_genes)) > 0
dds_genes <- dds_genes[keep_genes, ]
dds_genes <- estimateSizeFactors(dds_genes)

dds_genes%>%
group_by(.sample, sizeFactor, condition) %>%
summarize(library_size=sum(counts), .groups='drop') %>%
ggplot(aes(x=library_size, y=sizeFactor, color=condition)) +
geom_point(size=3) +
labs(x="Library size", y="Size factor", color = "Treatment") +
    scale_color_manual(values = c("CRISPR_CTRL" = "#3182bd", "CRISPR_KD" = "#9ecae1", "siRNA_CTRL" = "#c51b8a", "siRNA_KD" = "#fa9fb5"))
```

### Normalize with VST
```{r vst}
dst_vst <- vst(dds_genes, blind = TRUE)
assay(dds_genes, "vst_norm_counts") <- assay(dst_vst)

msd_plot <- vsn::meanSdPlot(assay(dds_genes, "vst_norm_counts"), ranks=FALSE, plot = FALSE)
msd_plot$gg +
labs(title="Sequencing depth normalized log2(read counts)",
x="Mean", y="Standard deviation")

dds_genes %>% 
  ggplot(aes(x=vst_norm_counts, group=.sample, color=condition)) + 
  geom_density() + 
  labs(x="Log normalized counts", y="Density (genes)", color="Genotype")
```

## Differential gene expression analysis

### Preparing for DE
```{r dgeAnalysis}
dds_genes$condition
# This is comparing everything to CRISPR_CTRL
dds_genes %<>% DESeq()
rowData(dds_genes) %>% colnames
rowData(dds_genes)$WaldPvalue_condition_siRNA_CTRL_vs_CRISPR_CTRL %>% hist(main = "siRNA_CTRL vs. CRISPR_CTRL", ylim = c(1, 10000))
rowData(dds_genes)$WaldPvalue_condition_CRISPR_KD_vs_CRISPR_CTRL %>% hist(main = "CRISPR_KD vs. CRISPR CTRL")
df_result_1 <- results(dds_genes, contrast = c("condition", "CRISPR_CTRL", "CRISPR_KD"), independentFiltering = TRUE, alpha = 0.05)
summary(df_result_1)
table(df_result_1$padj < 0.05)

df_result_2 <- results(dds_genes, contrast = c("condition", "CRISPR_CTRL", "siRNA_CTRL"), independentFiltering = TRUE, alpha = 0.05)
```

```{r dgeAssess}
df_result_1$padj %>% hist(main = "CRISPR_KD vs. CRISPR CTRL")
df_result_2$padj %>% hist(main = "siRNA_CTRL vs. CRISPR_CTRL", ylim = c(1, 10000))

df_result_1_sorted <- df_result_1 %>% `[`(order(.$padj),)
head(df_result_1_sorted)
plotCounts(dds_genes, gene = "ITGA6", normalized = TRUE, xlab = "")
plotCounts(dds_genes, gene = "METTL3", normalized = TRUE, xlab = "")

genes_dge <- rownames(subset(df_result_1_sorted, padj < 0.05))
vst_genes <- dst_vst[genes_dge,] %>% assay
pheatmap(vst_genes, scale="row",
show_rownames=FALSE, main="DGE (row-based z-score)")
```

### Relevel
In order to compare everything to siRNA_CTRL (e.g. compare siRNA_CTRL and siRNA_KD)
```{r relevel}
dds_genes$condition %<>% relevel(ref="siRNA_CTRL")
```
